### ISSUE TITLE
RocketChat 3.0.4 does not allow .zip uploads (HTTP 400)

### ISSUE THREAD

**Issue by ghost** [RocketChat/Rocket.Chat#17011]
Uploading of .pdf, .jpeg or .doc works but when I try to upload a ZIP archive, I get a HTTP 400 response back for the POST request from the back-end. Site URL is set correctly, I tried GridFS as well as local file-system, tried allow unauthroized uploads, etc. - this all did not help.

Btw I am running RocketChat via docker-compose in docker containers.

Any help is appreciated.

**Comment by 564064202** [comment]
Sir,my version 2.4.11 is working fine. You can try just as below
![image](https://user-images.githubusercontent.com/12368358/77614475-00449d00-6f68-11ea-8354-b854eb6a43cc.png)

Could you give me a hand.
https://github.com/RocketChat/Rocket.Chat/issues/17020

run docker-compose with multiple instances will meet a strange but ' MongoError: ns not found', 'Errors like this can cause oplog processing errors.'

**Comment by buchey8070** [comment]
Same issue here - without docker.
Yesterday updated to 3.0.x - Screenshots are working fine, zip-Uploads aren't. 

Configured same media types, increased upload size, nothing helped either.

**Comment by hirschnase** [comment]
Yes, same here. when I clear the list of allowed mime-types I can upload a ZIP file. Seems to be a mime-type mismatch.

**Comment by JeroenAdam** [comment]
Same here, uploading of .pdf works while .zip did not work. This issue for some users, while not for other users.
Solved by downgrading from v3.12 to v3.02.

**Comment by hans4rc** [comment]
same here

when i upload a zip file, i get the dialog correctly with no error message
![image](https://user-images.githubusercontent.com/66670106/84127291-5b524000-aa3f-11ea-9521-3a63d81b11eb.png)

but then i get this error message
![image](https://user-images.githubusercontent.com/66670106/84127409-8a68b180-aa3f-11ea-8363-b1f4c155c34c.png)

Upgrade from 2.4.11 to 3.3.0 last week 

I tested this mime types = application/x-zip and application/zip

**Comment by codeneno** [comment]
application/x-zip-comoressed

**Comment by hans4rc** [comment]
RC 3.3.0 - 1.15.0-beta.3411


Test with application/x-zip – Google Chrome, Edge, FireFox
![image](https://user-images.githubusercontent.com/66670106/85229653-54d2a980-b3eb-11ea-93ab-37d52226f2b3.png)

Test with application/x-zip-compressed – Google Chrome, Edge, FireFox
![image](https://user-images.githubusercontent.com/66670106/85229661-63b95c00-b3eb-11ea-99f6-72293b350117.png)


Test with application/zip – Google Chrome, Edge, FireFox
![image](https://user-images.githubusercontent.com/66670106/85229668-73d13b80-b3eb-11ea-989e-e072754dc5a2.png)

**When I put a zip in RC, everything still looks ok. The error message only appears when I click on Send/Upload. **

**Comment by codeneno** [comment]
sir,we as using 3.3.3 with application/x-zip-compressed,
every thing is fine

**Comment by hans4rc** [comment]
Hi codeneno, thanks for the reply, the error is reproducible on other computers. I will upgrade to 3.3.3

**Comment by codeneno** [comment]
you are welcome

**Comment by codeneno** [comment]
@hans4rc how many users do you have?

**Comment by hans4rc** [comment]
At present we have 146 users here.

**Comment by hans4rc** [comment]
Very interesting!
Upgraded to 3.3.3
Tested on various computers

application/zip = ok

application/x-zip = failed

application/x-zip-compressed = failed

application/zip,application/x-zip,application/x-zip-compressed = ok
application/zip,application/x-zip,application/x-zip-com**o**ressed = ok even with typing error

So, THE correct and only working MIME type is application/zip!

**Comment by codeneno** [comment]
haha,it's a small bug

**Comment by hans4rc** [comment]
At the moment i use version 3.4.2

I have set the following values for the compressed media types:
application/x-compressed-zip
application/x-rar-compressed
application/octet-stream
application/x-zip-compressed
application/zip
multipart/x-zip

And it works fine!

Thie issue can be closed.

**Comment by KevLehman** [comment]
This pull request has been inactive for a while, so we’re closing it to keep the backlog manageable.
If you’d like to continue working on it, feel free to reopen or submit a new PR. Thanks for the contribution!
